.github/workflows/checks.yaml

```diff
@@ -0,0 +1,74 @@
+name: Checks
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+env:
+  # 3.0 in quotes for https://github.com/actions/runner/issues/849
+  SUPPORTED_VERSIONS: '["3.0", 2.7, 2.6, 2.5]'
+
+jobs:
+  supported-versions:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+    - id: set-matrix
+      run: |
+        echo "::set-output name=matrix::${{env.SUPPORTED_VERSIONS}}"
+  unit-test:
+    needs: supported-versions
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby-version: ${{fromJson(needs.supported-versions.outputs.matrix)}}
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Ruby ${{ matrix.ruby-version }}
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby-version }}
+      - run: |
+          gem install bundler
+          bundle install
+      - run: rake test
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.7
```

Comment:
Would it make sense to also lint on version 3.x?

Reply:
Sure. Don't think it matters much